src/libs/actions/Report.js

```diff
@@ -379,11 +379,26 @@ function setLocalLastRead(reportID, sequenceNumber) {
 
     // Update the report optimistically
     Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
-        unreadActionCount: 0,
+        unreadActionCount: Math.max(reportMaxSequenceNumbers[reportID] - sequenceNumber, 0),
         lastVisitedTimestamp: Date.now(),
     });
 }
 
+/**
+ * Set the New marker for a given report to the oldest unread message
+ *
+ * @param {Number} reportID
+ */
+function setNewMarkerInitialPosition(reportID) {
+    const lastReadSequenceNumber = allReports[reportID].unreadActionCount === 0
```

Comment:
Please add comments to explain why the logic is this way (again, from an outsider, I have no idea why this logic is written this way which makes it incredibly difficult for me to maintain).